### ISSUE TITLE
Add support for gathering BlockVolume metrics from in-tree storage drivers

### ISSUE THREAD

**Issue by nixpanic** [kubernetes/kubernetes#101431]
#### What would you like to be added:

#97972 adds support for metrics from BlocMode volumes provided by CSI-drivers.

In-tree drivers can provide these metrics as well. At least the Capacity (size) of the block-device can be fetched with standard tools.

#### Why is this needed:

This makes it easier to test the changes from #97972 and can provide standard utilities that CSI-drivers can consume as well. An implementation should probably look like the `pkg/volume/metrics_statfs.go:NewMetricsStatFS()` call.

/sig storage
/assign nixpanic

**Comment by nixpanic** [comment]
/bug

Although #97972 is a standalone change for BlockVolumes, some of the in-tree storage drivers already provide the `MetricsProvider` interface on a type that gets extended by the (per driver) `NewBlockMapper()` call. This means that the `<blockMapper>.GetMetrics()` function is expected to be available, and the type casting in  #97972 is not sufficient.

Because of this, fetching metrics from some of the in-tree drivers will cause a panic:
```
panic(0x4203c20, 0x75ea6e0)
        /usr/lib/golang/src/runtime/panic.go:965 +0x1b9
k8s.io/kubernetes/pkg/volume/local.(*localVolumeMapper).GetMetrics(0xc003ce91c0, 0x4b2aa63, 0xf, 0x0)
```

**Comment by nixpanic** [comment]
/kind bug

See the description of the panic in the previous comment.
